code/game/objects/obj_defense.dm

```diff
@@ -4,9 +4,11 @@
 	if(QDELETED(src))
 		stack_trace("[src] taking damage after deletion")
 		return
+	if(obj_integrity <= 0)
+		stack_trace("[src] taking damage while having <= 0 integrity")
```

Comment:
if you don't early return here, it may cause more things to break by calling obj_destruction/obj_break multiple times.

Reply:
additionally you could just do a CRASH() but it's whatever